src/me.rs

```diff
@@ -52,18 +52,19 @@ pub fn motion_estimation(
       let range = 32 as isize;
       let blk_w = bsize.width();
       let blk_h = bsize.height();
-      let border_w = 128 + blk_w as isize * 8;
-      let border_h = 128 + blk_h as isize * 8;
-      let cols = (rec.frame.planes[0].cfg.width + MI_SIZE - 1) / MI_SIZE;
-      let rows = (rec.frame.planes[0].cfg.height + MI_SIZE - 1) / MI_SIZE;
-      let x_min = -(bo.x as isize) * (8 * MI_SIZE) as isize - border_w;
-      let x_max = (cols - bo.x - blk_w / MI_SIZE) as isize * (8 * MI_SIZE) as isize + border_w;
-      let y_min = -(bo.y as isize) * (8 * MI_SIZE) as isize - border_h;
-      let y_max = (rows - bo.y - blk_h / MI_SIZE) as isize * (8 * MI_SIZE) as isize + border_h;
-      let x_lo = po.x + ((-range).max(x_min / 8));
-      let x_hi = po.x + (range.min(x_max / 8));
-      let y_lo = po.y + ((-range).max(y_min / 8));
-      let y_hi = po.y + (range.min(y_max / 8));
+
+      let border_w = 16 + blk_w as isize; // in full pel unit
+      let border_h = 16 + blk_h as isize;
+
+      let x_lo = po.x - range;
+      let x_hi = po.x + range;
+      let y_lo = po.y - range;
+      let y_hi = po.y + range;
+
+      let x_lo = x_lo.max(-border_w);
+      let x_hi = x_hi.min((fi.w_in_b * MI_SIZE) as isize + border_w);
```

Comment:
This should include "- blk_w"?
I am not a fan of duplicating computations here. It should be best to define a MV range limit and then use this limit to determine the range for the initial ME

Reply:
Please see that I directly clip on x_hi (i.e. last search pixel on right, which is frame size aligned in MI_size + border_w, correct?) instead of range (i.e. offset).
Yes, I rephrased it for easier read, but x_hi = .. are duplicated thus can waste cpu cycles. 
Let me try how to not do that. (btw, pls see the trade-in that I used fi.w_in_b! :) )

Reply:
x_hi + blk_w should be less than fi.w_in_b * MI_SIZE + border_w, thus blk_w should be in there somewhere. The use of fi.w_in_b is good (had missed that previously)

Reply:
Oh, right- max ME search range should exclude block size itself.